ui/src/app/pages/admin/user/admin-user-profile.spec.tsx

```diff
@@ -434,94 +423,92 @@ describe('AdminUserProfile', () => {
+
+    expect(institutionalRoleDescriptionInput).toHaveValue(roleDetails);
 
-    saveButton = wrapper.find('[data-test-id="update-profile"]');
-    expect(saveButton.exists()).toBeTruthy();
-    expect(saveButton.props().disabled).toBeFalsy();
+    expectButtonElementEnabled(
+      screen.getByRole('button', {
+        name: /save/i,
+      })
+    );
   });
 
   it('should allow updating initial credit limit', async () => {
-    const wrapper = component();
-    expect(wrapper).toBeTruthy();
-    await waitOneTickAndUpdate(wrapper);
+    component();
+    await waitForNoSpinner();
 
-    expect(
-      findDropdown(wrapper, 'initial-credits-dropdown').props().value
-    ).toEqual(TARGET_USER_PROFILE.freeTierDollarQuota);
+    const initialCreditsDropdown = getDropdown('initial-credits-dropdown');
+
+    expect(initialCreditsDropdown.value).toEqual(
+      TARGET_USER_PROFILE.freeTierDollarQuota.toString()
+    );
 
     const newLimit = 800.0;
     expect(newLimit).not.toEqual(TARGET_USER_PROFILE.freeTierDollarQuota); // sanity check
 
-    await simulateComponentChange(
-      wrapper,
-      findDropdown(wrapper, 'initial-credits-dropdown'),
-      newLimit
-    );
-    expect(
-      findDropdown(wrapper, 'initial-credits-dropdown').props().value
-    ).toEqual(newLimit);
+    await user.click(initialCreditsDropdown);
+    await user.click(screen.getByText(`\$${newLimit.toFixed(2)}`));
+
+    expect(initialCreditsDropdown.value).toEqual(newLimit.toString());
 
-    const saveButton = wrapper.find('[data-test-id="update-profile"]');
-    expect(saveButton.exists()).toBeTruthy();
-    expect(saveButton.props().disabled).toBeFalsy();
+    expectButtonElementEnabled(
+      screen.getByRole('button', {
+        name: /save/i,
+      })
+    );
   });
 
   function expectModuleTitlesInOrder(
     accessModules: AccessModule[],
-    tableRows: ReactWrapper
+    tableRows: HTMLElement[]
   ) {
     accessModules.forEach((moduleName, index) => {
-      const moduleRow = tableRows.at(index);
+      const moduleRow = tableRows[index];
       const { adminPageTitle } = getAccessModuleConfig(moduleName);
-      expect(
-        findNodesContainingText(moduleRow, adminPageTitle).exists()
-      ).toBeTruthy();
+      expect(within(moduleRow).getByText(adminPageTitle)).toBeInTheDocument();
     });
   }
 
   it('should render the titles of all expected access modules', async () => {
-    const wrapper = component();
-    expect(wrapper).toBeTruthy();
-    await waitOneTickAndUpdate(wrapper);
-
-    const table = wrapper.find('[data-test-id="access-module-table"]');
-    expect(table.exists()).toBeTruthy();
+    component();
+    await waitForNoSpinner();
 
-    const tableRows = table.find('tbody tr[role="row"]');
-    expect(tableRows.length).toEqual(orderedAccessModules.length);
+    const table = screen.getByTestId('access-module-table');
+    const rows = within(table).getAllByRole('row');
+    rows.shift(); // remove the header row
```

Comment:
The previous version (line 520) is doing a css selector `tbody tr[role="row"]` so it was only getting the rows in the body of the table. Since css selectors are not the preferred way of selecting within RTL, I am relying on `getAllByRole('row');` which gets all the rows in the table (header and body). Since that is the case, we need to remove the first row since it is from the previously unaccounted for header row.

As an aside, there is a long open issue in the testing library that calls for a more intuitive way of testing tables:
https://github.com/testing-library/dom-testing-library/issues/583